pkg/awsutils/awsutils.go

```diff
@@ -353,19 +326,8 @@ func New(useCustomNetworking bool) (*EC2InstanceMetadataCache, error) {
 	cache.useCustomNetworking = useCustomNetworking
 	log.Infof("Custom networking %v", cache.useCustomNetworking)
 
-	sess, err := session.NewSession(
-		&aws.Config{
-			Region: aws.String(cache.region), 
-			MaxRetries: aws.Int(15),
-			HTTPClient: &http.Client{
-				Timeout: httpTimeoutValue,
-			},
-		},
-	)
-	if err != nil {
-		log.Errorf("Failed to initialize AWS SDK session %v", err)
-		return nil, errors.Wrap(err, "instance metadata: failed to initialize AWS SDK session")
-	}
+	awsCfg := aws.NewConfig().WithRegion(region)
```

Comment:
Can we move the "region" when we initialize "sess" ? That could avoid a copy.

Reply:
I thought of this. To move sess.copy to awssession and get region by querying ec2 metadata, get availability zone and then do some parsing did not seem efficient way, and more work.

Reply:
Can't afaics?  We need to find the region _somehow_.  Assuming we use IMDS to find the region, then we still need a 'regionless' session to initialise the IMDS client, so that we can _find_ the region, so that we can configure a 'regionful' session to initialise the 'normal' AWS client.

Reply:
you are right :D @anguslees